src/test/coins_tests.cpp

```diff
@@ -53,14 +53,11 @@ class CCoinsViewTest : public CCoinsView
 
     bool BatchWrite(CoinsViewCacheCursor& cursor, const uint256& hashBlock) override
     {
-        for (auto it{cursor.Begin()}; it != cursor.End(); it = cursor.NextAndMaybeErase(*it)){
-            if (it->second.IsDirty()) {
```

Comment:
is this is always the case, consider adding an assertion
```C++
assert(it->second.IsDirty());
```

(we can remove them in the last commit, but would be useful for the intermediary commits at least)

Reply:
Done.